src/impl.cxx

```diff
@@ -1152,11 +1151,9 @@ namespace ipr::impl {
 
          // Does the overload-set already contain a decl with that type?
          overload_entry* master = ovl->lookup(t);
-         impl::Typedecl* decl;
-         if (master == 0)       // no, this is the first declaration
-             decl = typedecls.declare(ovl, t);
-         else                   // just re-declare.
-            decl = typedecls.redeclare(master);
+         impl::Typedecl* decl = master == nullptr ?
```

Comment:
I can get easily confused by precedence in cases like this. I'd prefer having some extra parens.

Reply:
> I can get easily confused by precedence in cases like this. I'd prefer having some extra parens.

I would rather not have those extraneous parenthesis here.  Parenthesis are already overused in C++ (and C).